### PR TITLE
Page object model implementation

### PR DESCRIPTION
**Pull request by MrM0ra**

### CODE REVIEW

**Comment by garzuzo** [comment]
please fix conflict

File: cypress/page/menu-content.page.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    }
+
+    public visitMenuContentPage(): void {
+        cy.visit(this.menuContentPageURL)
+    }
+
+    public goToTShirtMenu(): void {
+        cy.get(this.tShirtMenu).click()
+    }
+
```
Comment (garzuzo) [inline cypress/page/menu-content.page.ts:13]: remove line

File: cypress/page/menu-content.page.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+}
+
```
Comment (garzuzo) [inline cypress/page/menu-content.page.ts]: remove line

File: cypress/page/payment-step.page.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+	public confirmOrder(): void {
+		cy.get(this.button).click()
+	}
+
+	public compare(): void {
+		cy.get(this.label).should('have.text', 'Your order on My Store is complete.');
+	}
+}
+
```
Comment (garzuzo) [inline cypress/page/payment-step.page.ts]: remove line

File: cypress/page/product-list.page.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	}
+
+	public selectTShirt(): void {
+		cy.get(this.tShirt).click()
+	}
+
+	public continue(): void {
+		cy.get(this.button).click()
+	}
+
```
Comment (garzuzo) [inline cypress/page/product-list.page.ts]: remove line

File: cypress/page/product-list.page.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+}
+
```
Comment (garzuzo) [inline cypress/page/product-list.page.ts]: remove line

File: cypress/page/shipping-step.page.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+	public check(): void {
+		cy.get(this.checkBox).click();
+	}
+
+	public proceed() :void {
+		cy.get(this.button).click();
+	}
+}
+
```
Comment (garzuzo) [inline cypress/page/shipping-step.page.ts]: remove line

File: cypress/page/shopping-cart.page.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+	private button: string;
+
+	constructor() {
+		this.button = ".cart_navigation span";
+	}
+
+	public proceed(): void {
+		cy.get(this.button).click()
+	}
+
```
Comment (garzuzo) [inline cypress/page/shopping-cart.page.ts]: remove line

File: cypress/page/shopping-cart.page.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+}
+
```
Comment (garzuzo) [inline cypress/page/shopping-cart.page.ts]: remove line

**Comment by garzuzo** [comment]
In general, the name of the methods could be more descriptive. e.g:
`shoppingCart.proceedToCheckoutSummary();` **is better than** `shoppingCart.proceed();`

**Review by garzuzo — CHANGES_REQUESTED** [review]
Please fix comments.

File: cypress/page/payment-step.page.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+	public selectBank(): void {
+		cy.get(this.bank).click()
+	}
+
+	public confirmOrder(): void {
+		cy.get(this.button).click()
+	}
+
+	public compare(): void {
+		cy.get(this.label).should('have.text', 'Your order on My Store is complete.');
```
Comment (garzuzo) [inline cypress/page/payment-step.page.ts]:
```suggestion
		cy.get(this.label);
```
The verification should be in the test

**Review by garzuzo — CHANGES_REQUESTED** [review]

File: cypress/page/payment-step.page.ts (hunk 9 above)
Comment (MrM0ra) [reply cypress/page/payment-step.page.ts]: So should i return the element so it can be verified on the test?

Comment (manuelq12) [reply cypress/page/payment-step.page.ts]: Exactly, the assertions should be done outside the page objects

File: cypress.json [hunk 10]
```diff
@@ -1,3 +1,5 @@
 {
-	"video": false
+	"video": false,
+	"defaultCommandTimeout": 20000,
+    "responseTimeout": 20000
```
Comment (manuelq12) [inline cypress.json]: Fix identation

File: cypress/integration/buy-shirt.spec.ts [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+describe("Buy a t-shirt", () => {
+  it("then should be bought a t-shirt", () => {
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+    productList.selectTShirt();
+    productList.proceedToCheckout();
+    shopping.proceedToCheckout();
+    login.typeEmail();
+    login.typePwd();
+    login.signIn();
```
Comment (manuelq12) [inline cypress/integration/buy-shirt.spec.ts]: This three can be one step, but it should be reusable, you should pass by parameter the user-password

File: cypress/page/login.page.ts [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+	private submit: string;
+
+	constructor(){
+		this.email = '#email';
+		this.password = '#passwd';
+		this.submit = '#SubmitLogin span';
+	}
+
+	public typeEmail(): void {
+		cy.get(this.email).type('aperdomobo@gmail.com')
```
Comment (manuelq12) [inline cypress/page/login.page.ts]: Parameters

File: cypress/page/login.page.ts [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+	}
+
+	public typePwd(): void {
+		cy.get(this.password).type('WorkshopProtractor')
+	}
```
Comment (manuelq12) [inline cypress/page/login.page.ts]: Parameters

**Review by manuelq12 — CHANGES_REQUESTED** [review]

**Review by manuelq12 — APPROVED** [review]
LGTM, great job!

**Review by garzuzo — APPROVED** [review]

**Comment by manuelq12** [comment]
Check for conflicts, not sure why there are conflicts

**Pull request merged**